capstone-backend/src/main/java/com/google/sps/servlets/Utility.java

```diff
@@ -316,4 +319,114 @@ else if (("remove_"+field.getName()).equals(key)) {
+      Field[] fields, List<String> requiredFields) throws IOException {
+    List<String> list = new ArrayList<>();
+    List<String> fieldNames = fields.stream().map(f -> f.getName()).collect(Collectors.toList());
+    list.addAll(jsonObject.keySet());
+    list.retainAll(fieldNames);
+    if (list.size() != jsonObject.keySet().size()) {
+      System.err.println("Error: Not all parameters in the request body are fields of the given class.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    list.clear();
+    list.addAll(jsonObject.keySet().stream()
+                    .filter(key -> jsonObject.get(key).getAsString().length() > 0)
+                    .collect(Collectors.toList()));
+    list.retainAll(requiredFields);
+    if (list.size() != requiredFields.size()) {
+      System.err.println("Error: Not all required fields are present in the request body, or some required fields are empty.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST);
+      return false;
+    } 
+    return true;
+  }
+
+  /**
+   * This method executes the given post request, creating the new object and returning it with its ID appended.
+   * @param {collectionReference} reference to the appropriate database collection
+   * @param {request} the request sent to the backend
+   * @param {genericClass} a Generic class, to be used in casting objects retrieved from the database
+   * @param {requiredFields} a list of the names of fields deemed "necessary" for creation of an object.
+   * @return T the object created by the post request, with an ID field generated by firebase
+   */
+  public static <T extends BaseEntity> T post(CollectionReference collectionReference, HttpServletRequest request, 
+      HttpServletResponse response, GenericClass<T> genericClass, List<String> requiredFields) throws IOException {
+    Map<String, Object> constructorFields = new HashMap<>();
+    JsonObject jsonObject = Utility.createRequestBodyJson(request);
+    Field[] fields = genericClass.getMyType().getDeclaredFields();
+    if (!postErrorHandler(jsonObject, response, fields, requiredFields)) {
+      return null;
+    }
+    for (Field f : fields) {
+      String name = f.getName();
+      String type = f.getType().getName();
+      if (type.equals("int") || type.equals("java.lang.Integer")) {
+        if (jsonObject.has(name) && jsonObject.get(name).getAsString().length() != 0) {
+          constructorFields.put(name, jsonObject.get(name).getAsInt());
+        }
+        else {
+          constructorFields.put(name, -1);
+        }
+      }
+      else if (type.equals("java.lang.String")) {
+        if (jsonObject.has(name) && jsonObject.get(name).getAsString().length() != 0) {
+          constructorFields.put(name, jsonObject.get(name).getAsString());
+        }
+        else {
+          constructorFields.put(name, "");
+        }
+      }
+      else if (type.equals("boolean") || type.equals("java.lang.boolean")) {
```

Comment:
This is just an idea, but maybe this could be replaced with type.contains("boolean") which will match either one of "boolean" or "java.lang.boolean". If you're worried about casing, you could do type.toLowerCase().contains("boolean")

Applies to your other type checks as well.

Reply:
I could do that, but I think I want to leave it as is to make it clear how I'm checking. I think using the contains() could be more confusing than explicitly checking for both the primitive and Object type.